homeassistant/components/life360/__init__.py

```diff
@@ -30,182 +31,147 @@
-    # be done with that config entry or that account from configuration.yaml.
-    # But if the config entry was created by import and the account no longer
-    # exists in configuration.yaml, or if the password has changed, then delete
-    # that out-of-date config entry.
-    already_configured = []
-    for entry in hass.config_entries.async_entries(DOMAIN):
-        # Find corresponding configuration.yaml entry and its password.
-        password = None
-        for account in conf[CONF_ACCOUNTS]:
-            if account[CONF_USERNAME] == entry.data[CONF_USERNAME]:
-                password = account[CONF_PASSWORD]
-        if password == entry.data[CONF_PASSWORD]:
-            already_configured.append(entry.data[CONF_USERNAME])
-            continue
-        if (
-            not password
-            and entry.source == config_entries.SOURCE_IMPORT
-            or password
-            and password != entry.data[CONF_PASSWORD]
-        ):
-            hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
-
-    # Create config entries for accounts listed in configuration.
-    for account in conf[CONF_ACCOUNTS]:
-        if account[CONF_USERNAME] not in already_configured:
-            hass.async_create_task(
-                hass.config_entries.flow.async_init(
-                    DOMAIN,
-                    context={"source": config_entries.SOURCE_IMPORT},
-                    data=account,
-                )
-            )
+    hass.data.setdefault(DOMAIN, IntegData(config.get(DOMAIN)))
     return True
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up config entry."""
-    hass.data[DOMAIN]["apis"][entry.data[CONF_USERNAME]] = get_api(
-        entry.data[CONF_AUTHORIZATION]
-    )
+    hass.data.setdefault(DOMAIN, IntegData())
+
+    # Check if this entry was created when this was a "legacy" tracker. If it was,
+    # update with missing data.
+    if not entry.unique_id:
+        hass.config_entries.async_update_entry(
+            entry,
+            unique_id=entry.data[CONF_USERNAME].lower(),
+            options=DEFAULT_OPTIONS | hass.data[DOMAIN].cfg_options,
+        )
+
+    coordinator = Life360DataUpdateCoordinator(hass, entry)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN].coordinators[entry.entry_id] = coordinator
+
+    # Set up components for our platforms.
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload config entry."""
-    try:
-        hass.data[DOMAIN]["apis"].pop(entry.data[CONF_USERNAME])
-        return True
-    except KeyError:
-        return False
+    del hass.data[DOMAIN].coordinators[entry.entry_id]
+
+    # Unload components for our platforms.
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
```

Comment:
We should probably unload the platforms first before we delete the coordinator for the config entry.

Reply:
Ok, will do.